prov/gni/include/fi_ext_gni.h

```diff
@@ -57,6 +57,7 @@ typedef enum dom_ops_val { GNI_MSG_RENDEZVOUS_THRESHOLD,
 			   GNI_MAX_RETRANSMITS,
 			   GNI_ERR_INJECT_COUNT,
 			   GNI_MR_CACHE_LAZY_DEREG,
+			   GNI_MR_CACHE_UDREG,
```

Comment:
Do we ever think people might want the option for *no caching*?  Say for example a client chooses to implement their own cache they might not want our cache.  If so, maybe this should be GNI_MR_CACHE_TYPE (or something like that) and we could parse strings like "default", "none", "udreg", etc.